lang/cs/Org.Apache.REEF.Client/Common/JavaClientLauncher.cs

```diff
@@ -152,15 +153,28 @@ private string GetClientClasspath()
             var files = Directory.GetFiles(JarFolder)
                 .Where(x => (!string.IsNullOrWhiteSpace(x)))
                 .Where(e => Path.GetFileName(e).ToLower().StartsWith(ClientConstants.ClientJarFilePrefix))
-                .ToArray();
+                .ToList();
 
-            if (null == files || files.Length == 0)
+            if (files.Count == 0)
             {
                 Exceptions.Throw(new ClasspathException(
                     "Unable to assemble classpath. Make sure the REEF JAR is in the current working directory."), Logger);
             }
-            var classPath = string.Join(";", files);
-            return classPath;
+
+            var classpathEntries = new List<string>(_additionalClasspathEntries).Concat(files);
```

Comment:
Any synchronization issue while you are copying _additionalClasspathEntries to a new list and someone is calling AddToClassPath()?